DataBindings/Botify
# Refactor unit tests

### Description
This pull request introduces significant improvements to the testing infrastructure of our project. We have refactored the existing unit tests and added new tests for critical functionality related to Spotify and Discord integration. These changes aim to enhance code quality, ensure reliable functionality, and improve maintainability.

### Changes Made
1. **Reorganized Test Suites**: We have organized the test suites to align with the project's structure and to ensure a clear separation of concerns. This makes it easier to locate and run specific tests.

2. **Enhanced Mocking with `jest-fetch-mock`**: To prevent actual HTTP requests during testing, we have employed `jest-fetch-mock` for mocking network requests. This ensures a controlled testing environment and minimizes external dependencies.

3. **Standardized Error Messages**: Error messages in the tests have been standardized, providing a consistent format and making it simpler to diagnose failures.

4. **Environment Variable Usage**: We have transitioned to utilizing environment variables for credentials and tokens, enhancing flexibility for testing in different environments and reducing hardcoding.

5. **Improved Data Validation**: In some tests, we have enhanced data validation to ensure a closer match between expected and actual data structures.

6. **Refactored Test Data**: We've updated test data to better reflect real-world scenarios, which results in more robust and realistic testing.

7. **Commented and Documented**: We have added comments and documentation to the test code, improving clarity and adherence to coding standards.

### Functions Tested
We have focused on testing functions related to Spotify and Discord:

#### Spotify
1. `getAccessToken`: Covered both success and failure scenarios.
2. `getUserPlaylists`: Tested success and error cases.
3. `searchSong`: Comprehensive testing, including success and failure scenarios.

#### Discord
1. `getDiscordToken`: Testing for both success and failure cases.

### Motivation and Context
This PR significantly elevates the quality of our unit testing suite, ensuring that essential functionality related to Spotify and Discord is thoroughly verified. By standardizing error messages, using environment variables, and enhancing data validation, we create a more reliable testing environment. This, in turn, contributes to a more robust and maintainable codebase.

### How to Test
To validate these changes:

1. Ensure you have the necessary environment variables set correctly (e.g., `SPOTIFY_CLIENT_ID`, `SPOTIFY_CLIENT_SECRET`, `DISCORD_TOKEN`).
2. Run the provided unit tests using your preferred testing framework (e.g., Jest) to verify the functionality of the tested functions.

### Checklist
- [x] Reorganized test suites for better organization.
- [x] Implemented `jest-fetch-mock` for reliable HTTP request mocking.
- [x] Standardized error messages for consistent failure diagnosis.
- [x] Utilized environment variables for credentials and tokens.
- [x] Improved data validation for more accurate testing.
- [x] Updated test data to reflect real-world scenarios.
- [x] Added comments and documentation for clarity and adherence to coding standards.

### Related Issues
N/A
